### ISSUE TITLE
FNX3-16654 ⁃ [Bug] Crash loop on launch when device has no connection (e.g, Airplane mode)

### ISSUE THREAD

**Issue by AaronMT** [mozilla-mobile/fenix#2888]
Build: 0ca317b7ca9a739ecfa5b8d9363f4e7f06bad72f
Device: Pixel 2

(CC @eoger )

See also: https://github.com/mozilla-mobile/fenix/issues/2810 for another network crash

```java
2019-05-27 12:10:08.653 10104-10104/org.mozilla.fenix.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.debug, PID: 10104
    mozilla.appservices.fxaclient.FxaException$Network: Network error: Network error: java.io.IOException: "org.mozilla.geckoview.WebRequestError: Request failed, error=0x25, category=0x5"
        at mozilla.appservices.fxaclient.rust.RustError.intoException(RustError.kt:43)
        at mozilla.appservices.fxaclient.FirefoxAccount.ensureCapabilities(FirefoxAccount.kt:717)
        at mozilla.components.service.fxa.FxaDeviceConstellation$ensureCapabilitiesAsync$1.invokeSuspend(FxaDeviceConstellation.kt:88)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:238)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742)

**Comment by pocmo** [comment]
Will get fixed by:
https://github.com/mozilla-mobile/android-components/issues/1390

**Comment by grigoryk** [comment]
Actually, this is the correct a-c issue that will deal with this: mozilla-mobile/android-components#2724

**Comment by grigoryk** [comment]
Should be fixed now that we're on 0.55.0 of a-c, as of https://github.com/mozilla-mobile/fenix/pull/2802.
